### PR TITLE
feat: 🎸 add/remove agent to a PermissionGroup

### PR DESCRIPTION
**Pull request by sansan**
### JIRA Link 

https://polymesh.atlassian.net/browse/DA-94
https://polymesh.atlassian.net/browse/DA-95

### Changelog / Description 

- adds endpoint to list grous for asset (with permissions assigned to the group)
- adds endpoint to add an agent to a group (group can be passed as id, known group type or permissions)
- adds endpoint to remove external agent

### Checklist - 

- [x] New Feature ?
- [x] Updated swagger annotation (if API structure is changed) ?
- [x] Unit Test (if possible) ?

### CODE REVIEW

File: src/assets/models/permission-group-with-permissions.model.ts [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+import { ApiProperty, ApiPropertyOptional } from '@nestjs/swagger';
+import { PermissionGroupType } from '@polymeshassociation/polymesh-sdk/types';
+import { BigNumber } from 'bignumber.js';
+import { Type } from 'class-transformer';
+
+import { createGroupPermissionsModel } from '~/assets/assets.util';
+import { GroupPermissionsModel } from '~/assets/models/group-permissions.model';
+
+export class PermissionGroupWithPermissionsModel {
+  @ApiPropertyOptional({
```
Comment (polymath-eric) [inline src/assets/models/permission-group-with-permissions.model.ts:11]: Shouldn't the model always have an ID? Why is this optional?

File: src/assets/models/permission-group-with-permissions.model.ts [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    description: 'The ID of the Permission Group',
+    type: 'string',
+    format: 'bigint',
```
Comment (polymath-eric) [inline src/assets/models/permission-group-with-permissions.model.ts:14]: `example` ?

I think enum and type might not need them specified

File: src/assets/dto/invite-agent-to-group.dto.ts [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+import { TransactionBaseDto } from '~/common/dto/transaction-base-dto';
+
+export class PermissionsDto extends PickType(CreatePermissionGroupDto, [
+  'transactions',
+  'transactionGroups',
+] as const) {}
+
+export class InviteAgentToGroupDto extends TransactionBaseDto {
+  @ApiProperty({
+    description: 'The did of the target identity to invite to the group',
```
Comment (prashantasdeveloper) [inline src/assets/dto/invite-agent-to-group.dto.ts:18]:
```suggestion
    description: 'The DID of the target identity to invite to the group',
```

File: src/assets/dto/remove-agent-from-grop.dto.ts [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import { OmitType } from '@nestjs/swagger';
```
Comment (prashantasdeveloper) [inline src/assets/dto/remove-agent-from-grop.dto.ts:1]: istanbul ignore

File: src/assets/models/group-permissions.model.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+      'Transactions that the Account can execute. A null value represents full permissions',
+    type: TransactionPermissionsModel,
+    nullable: true,
+  })
+  @Type(() => TransactionPermissionsModel)
+  readonly transactions: TransactionPermissionsModel | null;
+
+  @ApiProperty({
+    description:
+      'Transaction Groups that the Account can execute. Having permissions over a [TxGroup](https://github.com/polymeshassociation/polymesh-sdk/blob/docs/v14/docs/enums/txgroup.md) means having permissions over every TxTag in said group. Note if `transactions` is null, ignore this value',
```
Comment (prashantasdeveloper) [inline src/assets/models/group-permissions.model.ts:20]: the url gives 404

File: src/assets/models/permission-group-with-permissions.model.ts [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+  })
+  readonly id?: BigNumber;
+
+  @ApiPropertyOptional({
+    description: 'The type of the Permission Group',
+    enum: PermissionGroupType,
+  })
```
Comment (prashantasdeveloper) [inline src/assets/models/permission-group-with-permissions.model.ts:21]: @IsOptional and @IsEnum missing

File: src/assets/assets.controller.ts [hunk 7]
```diff
@@ -651,7 +655,7 @@ export class AssetsController {
       'The signing identity does not have the required permissions to create a permission group',
     ],
   })
-  @Post(':asset/create-permission-group')
+  @Post(':asset/permission-groups/create')
```
Comment (prashantasdeveloper) [inline src/assets/assets.controller.ts:658]: If the endpoint is renamed, this should be marked as breaking change

File: src/assets/assets.controller.ts [hunk 8]
```diff
@@ -671,4 +675,83 @@ export class AssetsController {
...
+    ],
+  })
+  @Post(':asset/permission-groups/remove-agent')
+  public async removeAgent(
+    @Param() { asset }: AssetParamsDto,
+    @Body() params: RemoveAgentFromGroupDto
+  ): Promise<TransactionResponseModel> {
+    const result = await this.assetsService.removeAgentFromAsset(asset, params);
+
+    return handleServiceResult(result);
```
Comment (prashantasdeveloper) [inline src/assets/assets.controller.ts:755]: All these APIs should be moved in its own controller - PermissionGroupController with endpoint as :asset/permission-groups

**Review by prashantasdeveloper — CHANGES_REQUESTED** [review]
I think we should have separate permission-group module, controller, service so that we would not overstuff asset classes.

**Review by polymath-eric — APPROVED** [review]

**Pull request merged**